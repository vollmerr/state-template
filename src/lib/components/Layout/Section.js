import React from 'react';
import classNames from 'classnames';
import T from 'prop-types';

import Wrapper from './Wrapper';

const Section = (props) => {
  const { variant, ...rest } = props;
  const baseClass = classNames([
    'section',
    { [`section-${variant}`]: variant },
  ]);

  return (
    <Wrapper baseClass={baseClass} {...rest} />
  );
};

Section.propTypes = {
  variant: T.oneOf([null, 'default', 'standout', 'understated']),
};

Section.defaultProps = {
  variant: null,
};

export default Section;
