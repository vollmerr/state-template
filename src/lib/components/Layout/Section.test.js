import React from 'react';
import { shallow } from 'enzyme';

import { Section as fromLib } from '../..';
import Section from './Section';

const props = {
  variant: 'standout',
  className: 'test-class',
};

let wrapper;
describe('Section', () => {
  beforeEach(() => {
    wrapper = shallow(<Section {...props} />);
  });

  it('should be exported as part of the library', () => {
    expect(fromLib).toEqual(Section);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should use the variant in the base class', () => {
    expect(wrapper.prop('baseClass')).toMatch(`section-${props.variant}`);
  });
});
