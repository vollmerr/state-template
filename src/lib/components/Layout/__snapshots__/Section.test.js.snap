// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render correctly 1`] = `
<LayoutBase
  baseClass="section section-standout"
  className="test-class"
  tag="div"
>
  test-children
</LayoutBase>
`;
